internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -99,10 +106,38 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 		"location": commonschema.Location(),
 
 		"capacity": {
-			Type:         pluginsdk.TypeInt,
-			Optional:     true,
-			Default:      20,
-			ValidateFunc: validation.IntPositive,
+			Type:          pluginsdk.TypeInt,
+			Optional:      true,
+			ConflictsWith: []string{"auto_scale_profile"},
```

Comment:
If the desired behaviour is `ExactlyOneOf` then we usually wrap this around the `features.FourPointOh` flag or at the very least leave a `TODO 4.0` comment so that maintainers know a breaking change needs to be introduced here for the next major release 
```suggestion
			// TODO 4.0 replace this with ExactlyOneOf
			ConflictsWith: []string{"auto_scale_profile"},
```

Reply:
👍 Thanks for sharing this. After an internal chat within our team, I think we'd like to leave the breaking change off entirely for now as we are also evaluating other changes as well which might impact some behavior in Terraform. 

Given it's unclear from our end around what those changes might be and the associated timelines, this comment might be confusing and I going to leave it off for the time being.